src/app/clusters/webrtc-transport-provider-server/webrtc-transport-provider-server.cpp

```diff
@@ -58,6 +58,153 @@ NodeId GetNodeIdFromCtx(const CommandHandler & commandHandler)
     return descriptor.subject;
 }
 
+/**
+ * @brief Validates spec-level SFrame constraints (data model constraints).
+ *
+ * Checks that SFrameConfig meets the data model requirements:
+ * - CipherSuite >= 1
+ * - BaseKey length <= 128 bytes
+ * - KID length 2-8 bytes
+ *
+ * @param[in] sframeConfig The SFrame configuration to validate
+ *
+ * @return true if all spec constraints are satisfied, false otherwise.
+ */
+bool SFrameFollowsSpecConstraints(const Clusters::WebRTCTransportProvider::Structs::SFrameStruct::DecodableType & sframeConfig)
+{
+    // Spec constraint: CipherSuite >= 1
+    if (sframeConfig.cipherSuite < 1)
+    {
+        return false;
+    }
+
+    // Spec constraint: BaseKey length <= 128
+    if (sframeConfig.baseKey.size() > 128)
+    {
+        return false;
+    }
+
+    // Spec constraint: KID length must be 2-8
+    if (sframeConfig.kid.size() < 2 || sframeConfig.kid.size() > 8)
+    {
+        return false;
+    }
+
+    return true;
+}
+
+/**
+ * @brief Checks if a URL has a turns or stuns scheme.
+ *
+ * @param[in] url The URL to check
+ *
+ * @return true if the URL starts with "turns:" or "stuns:", false otherwise.
+ */
+bool HasTurnsOrStunsScheme(const CharSpan & url)
+{
+    // Convert CharSpan to string for easier comparison
+    std::string urlStr(url.data(), url.size());
+
+    // Check for "turns:" or "stuns:" prefix (case-sensitive per URL spec)
+    return (urlStr.size() >= 6 && urlStr.substr(0, 6) == "turns:") || (urlStr.size() >= 6 && urlStr.substr(0, 6) == "stuns:");
+}
+
+/**
+ * @brief Validates ICEServers list constraints (data model constraints).
+ *
+ * Checks that ICEServers list meets the data model requirements:
+ * - ICEServers list: max 10 entries
+ * - URLs list per ICEServerStruct: max 10 items, each URL max 2000 characters
+ * - Username: optional, max 508 bytes
+ * - Credential: optional, max 512 bytes
+ * - CAID: optional, range 0-65534
+ *
+ * @param[in] iceServers The ICEServers list to validate
+ * @param[in] commandName Name of the command (for logging)
+ *
+ * @return Status::Success if all constraints are satisfied
+ *         Status::ConstraintError if a constraint is violated
+ *         Status::InvalidCommand if the list structure is invalid
+ */
+Status ICEServersFollowsSpecConstraints(
+    const DataModel::DecodableList<Clusters::Globals::Structs::ICEServerStruct::DecodableType> & iceServers,
```

Comment:
So why is this using `ICEServerStruct::DecodableType` while `CheckTurnsOrStunsRequiresUTCTime` uses `ICEServerDecodableStruct`?

Reply:
Acked